proposals/nnnn-main-menu-updating.md

```diff
@@ -0,0 +1,182 @@
+
+### RPC API Updates
+We're going to need to update `AddCommand` and `AddSubmenu` and add a few new RPC notifications to support dynamic updating artworks and sub-menus.
+
+###### New RPC Notifications
+In order to support this feature, new notifications have to be enabled from the head unit to the app library to indicate which menu items need updating.
+
+```xml
+<function name="UpdateCommand" functionID="UpdateCommandID" messagetype="notification" since="x.x">
+    <description>This notification tells an app to update the AddCommand with the requested data</description>
+
+    <param name="cmdID" type="Integer" minvalue="0" maxvalue="2000000000" mandatory="true">
+        <description>This cmdID must match a cmdID in the current menu structure.</description>
+    </param>
+
+    <param name="updateArtwork" type="Bool" mandatory="false">
+        <description>If not set, assume false. If true, the app should send a new AddCommand with the same cmdID as the previous one with the cmdIcon set. This will replace the UI cell with a new cell with the artwork displayed.</description>
+    </param>
+</function>
+```
+
+```xml
+<function name="UpdateSubMenu" functionID="UpdateSubMenuID" messagetype="notification" since="x.x">
+    <description>This notification tells an app to update the AddSubMenu or its 'sub' AddCommand and AddSubMenus with the requested data</description>
+
+    <param name="menuID" type="Integer" minvalue="0" maxvalue="2000000000" mandatory="true">
+        <description>This menuID must match a menuID in the current menu structure</description>
+    </param>
+
+    <param name="updateArtwork" type="Bool" mandatory="false">
+        <description>If not set, assume false. If true, the app should send a new AddSubMenu with the same menuID as the previous one with the menuIcon set. This will replace the UI cell with a new cell with the artwork displayed. If subcells have already been sent, they should be transferred to the new AddSubmenu cell (and don't need to be re-sent).</description>
+    </param>
+
+    <param name="updateSubCells" type="Bool" mandatory="false">
+        <description>If not set, assume false. If true, the app should send AddCommands with parentIDs matching the menuID. These AddCommands will then be attached to the submenu and displayed if the submenu is selected.</description>
+    </param>
+</function>
+```
+
+#### Capabilities Updates
+Updates will need to be made so that the app libraries and developers know if the head unit supports this feature or if they need to stick to the old-style of sending the entire menu at once.
+
+```xml
+<struct name="WindowCapability" since="6.0">
+    <!-- New Parameters -->
+    <param name="supportsDynamicMenus" type="Bool" mandatory="false">
+        <description>If true, the head unit will send UpdateCommand and UpdateSubMenu notifications and supports updating menu cells by sending the same cmdID / menuID with new data. If not set, assume false.</description>
```

Comment:
Please update to this:  

If true, the head unit will send UpdateCommand and UpdateSubMenu notifications and support updating menu cells by sending the same cmdID / menuID with new data. If not set, assume false.

Reply:
"supports" is correct here. It is referencing "head unit" not "notifications", so it's singular.

Reply:
Changed to make the sentence clearer